### PR TITLE
RAIL-4113 Add the InsightList component

### PR DESCRIPTION
**Pull request by krejzd**
Add the InsightList component
JIRA: RAIL-4113 

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8195/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10035/ : FAILURE in 5m 53s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8217/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10062/ : FAILURE in 8m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8223/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10067/ : SUCCESS in 11m 06s

File: libs/sdk-ui-dashboard/src/presentation/insightList/InsightList.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    const intl = useIntl();
+
+    const backend = useBackendStrict();
+    const workspaceId = useWorkspaceStrict();
+    const [initialLoadCompleted, setInitialLoadCompleted] = useState(false);
+    const [pagesToLoad, setPagesToLoad] = useState<number[]>([0]); // first page loaded
+    const [search, setSearch] = useState("");
+    const [selectedTabId, setSelectedTabId] = useState(INSIGHT_DROPDOWN_MY_TAB_ID);
+    const userRef = useDashboardSelector(selectCurrentUserRef);
+    const userUri = "uri" in userRef ? userRef.uri : undefined;
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/insightList/InsightList.tsx:58]: We have isUriRef in idk-model for this check, please use that.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8242/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10092/ : SUCCESS in 10m 08s

**Review by no23reason — APPROVED** [review]
LGTM

**Pull request merged**